tests/fixtures/datasource.py

```diff
@@ -18,6 +18,7 @@
 datasource_post = {
     "id": None,
     "column_formats": {"ratio": ".2%"},
+    "database": {"id": 1},
```

Comment:
Eventually it would be good to move these to randomly generated IDs (note to self)

Reply:
Yes, actually here I replace it in the test itself. Not sure what the proper approach here, a function that receives an object and returns a fixture? I'm guessing with DAO abstraction we can get more creative on generating proper fixtures.